client/actions/events/api.ts

```diff
@@ -628,74 +641,25 @@ const save = (original, updates) => (
                 EVENTS.UPDATE_METHODS[0].value :
                 eventUpdates.update_method?.value ?? eventUpdates.update_method;
 
-            const [planningsToCreate, planningsToUpdate] = partition(
-                (eventUpdates.associated_plannings ?? []).map((planning) => planningUtils.modifyForServer(planning)),
-                (item) => item._id.startsWith(TEMP_ID_PREFIX),
-            );
-
-            // ensure `associated_plannings` doesn't contain `planningsToCreate`
-            // otherwise it would get created twice - separately and together with event patch
-            eventUpdates.associated_plannings = planningsToUpdate;
-
             const createOrUpdatePromise: Promise<Array<IEventItem>> = originalEvent?._id != null ?
                 planningApi.events.update(originalItem, eventUpdates) :
                 planningApi.events.create(eventUpdates);
 
             return createOrUpdatePromise.then(([updatedEvent]: Array<IEventItem>) => {
-                if (updates.associated_plannings == null) {
-                    return Promise.resolve([updatedEvent]);
+                if (updatedEvent?._id != null) {
+                    updateLinkedPlanningsForEvent(
+                        updatedEvent._id,
+                        updates.associated_plannings,
+                    ).then((updatedPlannings) => {
+                        // Update associated events, so if a link has been added/removed etags are updated
+                        // after the change in planning items
+                        updatedEvent.associated_plannings = updatedPlannings;
+
+                        return [updatedEvent];
+                    });
+                } else {
+                    return [updatedEvent];
                 }
```

Comment:
Minor suggestion. You could invert the if statement and have an early return and eliminate the need for an else `block`. What do you think?

```suggestion
                if (updatedEvent?._id == null)
                    return [updatedEvent];
                
                return updateLinkedPlanningsForEvent(
                    updatedEvent._id,
                    updates.associated_plannings,
                ).then((updatedPlannings) => {
                    // Update associated events, so if a link has been added/removed etags are updated
                    // after the change in planning items
                    updatedEvent.associated_plannings = updatedPlannings;
                    return [updatedEvent];
                });
```